airbnb_project/listings/tasks.py

```diff
@@ -1,37 +1,50 @@
 from celery import shared_task
-from twisted.internet import reactor
-from scrapy.crawler import CrawlerRunner
+from scrapy.crawler import CrawlerProcess
 from scrapy.utils.log import configure_logging
 from listings.harvester_app.harvester.spiders.listings_spider import ListingsSpider
 from listings.harvester_app.harvester.harvester_settings import get_harvester_settings
 import logging
 
-# Set up logger for the Celery task
 logger = logging.getLogger(__name__)
 
-@shared_task(bind=True)
-def run_harvest_task(self):
+configure_logging()
+
+
+def run_spider():
     """
-    Celery task to run the Scrapy spider.
+    Run the Scrapy spider for harvesting listings.
+
+    This function initializes a Scrapy CrawlerProcess with the required settings,
+    schedules the `ListingsSpider` to run, and starts the crawling process.
+    The function runs in non-blocking mode with `stop_after_crawl=False` to keep
+    the process active after the spider completes.
+
+    Returns:
+        None
     """
-    try:
-        # Configure Scrapy logging
-        configure_logging()
+    runner = CrawlerProcess(settings=get_harvester_settings())
+    runner.crawl(ListingsSpider)
+    runner.start(stop_after_crawl=False)
 
-        # Set up the Scrapy crawler with custom settings
-        runner = CrawlerRunner(settings=get_harvester_settings())
 
-        # Start the crawling process
-        deferred = runner.crawl(ListingsSpider)
+@shared_task(bind=True)
```

Comment:
should we add max_retries here now that we aren't doing a retry in catch block?

Reply:
I was thinking of skipping retries entirely, but if we do include them, it should be for specific exceptions. I'd need to identify which exceptions warrant a retry.